### PR TITLE
Base framework for a rust auth library

### PR DESCRIPTION
**Pull request by sai-sunder-s**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -36,6 +36,7 @@ members = [
   "src/generated/cloud/iam/v1",
   "src/generated/api",
   "src/integration-tests",
+  "src/auth2"
```
Comment (coryan) [inline Cargo.toml:39]: src/auth is reserved for you.  `auth/` is where the old code lives.

**Comment by coryan** [comment]
Seems like this PR is now obsolete, can I close it?

**Comment by sai-sunder-s** [comment]
Yes it can be closed. Please do not delete the branch yet though. Bits and
pieces of code from here are being used to write the final code.

**Comment by dbolduc** [comment]
@sai-sunder-s - I deleted this branch in the `googleapis` repo. We should still be able to see the code. If you needed the branch for some reason, you can grab it from my fork: https://github.com/dbolduc/google-cloud-rust/tree/auth-framework-273-stash

**Pull request closed without merging**